components/style/parser.rs

```diff
@@ -74,3 +74,7 @@ pub fn log_css_error(input: &mut Parser, position: SourcePosition, message: &str
 pub trait Parse {
     fn parse(input: &mut Parser) -> Result<Self, ()> where Self: Sized;
 }
+
+pub trait ParseWithContext {
```

Comment:
This should be a temporary fix. We should probably replace all the types implementing the `Parse` method to take the context. You can do it if you like, but in case you don't, please have a FIXME above this, like, ` FIXME(waffles): Merge this with Parse`